### DISCUSSION TITLE
[Core Protocol] Optmism Mainnet Sequencer

### DISCUSSION THREAD

**Discussion by yohanelly95** [ethereum-optimism/developers#286] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [ ] Yes
- [X] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

We were running a few validators who are participating in a protocol deployed on OP-Sepolia. We got this error from the Alchemy RPC: `Optimism sequencer global transaction limit exceeded.`. This got us wondering if there were any rate limits if any on the Mainnet Sequencer, could this actually happen on OP Mainnet?  If so,

Under what circumstances if any would the global sequencer limit mainnet txns? And does how long does it take to reset the limit ?

How does L1 storage cost calculation factor in sequencer txn rate limiting if any? 

Does the mainnet gas fee handle this, if so it is fee based ordering of txns or first in first out?

Would the Ecotone network hardfork have any effect on this?

### Additional Information

The validators that our protocol will run will need 100% uptime, since downtime would be penalised. So we just want to be sure we handle any Global rate limiting by the sequencer if any.

**Comment by imtipi** [comment]
cc @sbvegan

**Comment by hash1go** [comment]
According to [Alchemy's documentation](https://docs.alchemy.com/reference/optimism-api-faq#what-is-the-transaction-throughput-on-optimism), throughput is set according to the grade of the account, which likely exists on the Mainnet as well. Please refer to [Alchemy's documentation](https://docs.alchemy.com/reference/optimism-api-faq#what-is-the-transaction-throughput-on-optimism) for details on the rate limits.

Also, as stated in [the documentation](https://docs.alchemy.com/reference/optimism-error-codes#429-errors), be sure to implement retries.

Rate limits are unrelated to on-chain costs. And the Ecotone hardfork does not have a direct relation to rate limits.

**Reply by yohanelly95** [reply]
I see, so this is rpc specific. Okay, so on OP Mainnet there are no sequencer rate limits as such, where can I read more about the specifications of the sequencer itself ?

Also, are the transactions processed by the sequencer based on fee based ordering or first in first out?

**Reply by hash1go** [reply]
> Where can I read more about the specifications of the sequencer itself ?

Sorry, but I couldn't find a document that explains this specification.


> Also, are the transactions processed by the sequencer based on fee based ordering or first in first out?

According to [this document](https://docs.optimism.io/stack/explainer#configurable-sequencer-per-op-chain), the OP Stack allows developers to set a specification based on FIFO for transaction sequencing. However, I couldn’t find the specification for the OP Mainnet itself.
From some sample investigations I conducted using the OP Mainnet explorer, the order of transactions within a block seems to be based on the amount of the priority fee. Since Optimism follows the EIP-1559 fee model, for a transaction to be included in a block, it must have a gas price set higher than the base fee.

**Comment by sbvegan** [comment, marked as answer]
> Under what circumstances if any would the global sequencer limit mainnet txns? And does how long does it take to reset the limit ?

There is a rate limit, but we don't publish that information because its subject to change and we don't want people designing their application to handle this in a static way.

> How does L1 storage cost calculation factor in sequencer txn rate limiting if any?

The storage cost is not a factor.

> Does the mainnet gas fee handle this, if so it is fee based ordering of txns or first in first out?

Here are some [documentation](https://docs.optimism.io/builders/dapp-developers/transactions/fees) on the gas fees. The OP Stack is Ethereum Equivalent, meaning it also has a EIP-1559 fee mechanism. So its not FIFO, but based on priority fees.

> Would the Ecotone network hardfork have any effect on this?

The [fee forumla](https://docs.optimism.io/builders/dapp-developers/transactions/fees#ecotone) is different in Ecotone, but it doesn't change from a user's perspective.

**Comment by yohanelly95** [comment]
> > Under what circumstances if any would the global sequencer limit mainnet txns? And does how long does it take to reset the limit ?
> 
> There is a rate limit, but we don't publish that information because its subject to change and we don't want people designing their application to handle this in a static way.
> 
> > How does L1 storage cost calculation factor in sequencer txn rate limiting if any?
> 
> The storage cost is not a factor.
> 
> > Does the mainnet gas fee handle this, if so it is fee based ordering of txns or first in first out?
> 
> Here are some [documentation](https://docs.optimism.io/builders/dapp-developers/transactions/fees) on the gas fees. The OP Stack is Ethereum Equivalent, meaning it also has a EIP-1559 fee mechanism. So its not FIFO, but based on priority fees.
> 
> > Would the Ecotone network hardfork have any effect on this?
> 
> The [fee forumla](https://docs.optimism.io/builders/dapp-developers/transactions/fees#ecotone) is different in Ecotone, but it doesn't change from a user's perspective.

Regarding the rate limiting on the global sequencer, the nodes that we will run for our protocol should ideally have 100% uptime since any downtime will be flagged as non participation in the protocol and they maybe subject to slashing. So if there is an approximate that we could go by, such that we are sure the nodes wont hit the limit or close to it. @sbvegan
